test: crawlerFiles path-traversal test assumes /etc/passwd exists (fails on Windows)

## Summary

One test in `app/api/src/routes/crawlerFiles.uploads.test.js` fails on Windows because it asserts that `/etc/passwd` exists as a sentinel to prove a delete operation didn't escape the config folder. That path does not exist on Windows.

## Failing test

```
crawler-configs/:configId/files — real upload/list/delete cycle
  > strips directory components from a path-traversal filename instead of escaping the config folder
```

## Error

```
AssertionError: expected false to be true // Object.is equality

  expect(existsSync('/etc/passwd')).toBe(true); // the real file, untouched
```

Line 112 of `crawlerFiles.uploads.test.js`.

## Root cause

The test verifies that a path-traversal filename like `../../etc/passwd` is sanitised before the delete, and that the real `/etc/passwd` was not deleted. The assertion `expect(existsSync('/etc/passwd')).toBe(true)` is Linux-only. On Windows the path doesn't exist, so `existsSync` returns `false` and the assertion fails — even though the security behaviour being tested is correct.

## Fix

Replace the `/etc/passwd` sentinel with a platform-neutral approach. For example, create a known decoy file in a temp directory before the test, attempt the traversal-delete, then assert the decoy still exists:

```js
import { mkdtempSync, writeFileSync } from 'fs';
import { join } from 'path';
import os from 'os';

const decoyDir = mkdtempSync(join(os.tmpdir(), 'ia-test-'));
const decoyFile = join(decoyDir, 'sentinel.txt');
writeFileSync(decoyFile, 'should not be deleted');

// ... perform the traversal-delete request ...

expect(existsSync(decoyFile)).toBe(true); // traversal did not escape
```

Or simply remove the `/etc/passwd` check entirely — the existing assertion that `res.body.deleted === 'passwd'` and that the decoy inside the config folder was deleted is sufficient to prove the path was sanitised without needing to verify an OS file.

## Affected file

- `app/api/src/routes/crawlerFiles.uploads.test.js` (1 test)

--- a/app/api/src/routes/crawlerFiles.uploads.test.js
+++ b/app/api/src/routes/crawlerFiles.uploads.test.js
@@ -97,7 +97,11 @@ describe('crawler-configs/:configId/files — real upload/list/delete cycle', ()
     // basename() reduces '../../etc/passwd' to just 'passwd' — confirm the
     // delete can only ever touch a file inside this config's own folder, by
     // planting a decoy 'passwd' file there and confirming only THAT gets
-    // removed — the real /etc/passwd is never touched.
+    // removed. Together these assertions prove containment: had the traversal
+    // escaped, `deleted` would not be the bare 'passwd' and the in-folder decoy
+    // would survive. (We deliberately don't assert on a real OS file like
+    // /etc/passwd — that sentinel is Linux-only and the checks below already
+    // prove the path was sanitised, portably.)
     const configDir = join(UPLOAD_ROOT_DIR, 'csv-501');
     const decoy = join(configDir, 'passwd');
     mkdirSync(configDir, { recursive: true });
@@ -107,9 +111,8 @@ describe('crawler-configs/:configId/files — real upload/list/delete cycle', ()
     const res = await request(makeApp())
       .delete('/api/admin/crawler-configs/501/files/' + encodeURIComponent('../../etc/passwd'));
     expect(res.status).toBe(200);
-    expect(res.body.deleted).toBe('passwd');
-    expect(existsSync(decoy)).toBe(false);
-    expect(existsSync('/etc/passwd')).toBe(true); // the real file, untouched
+    expect(res.body.deleted).toBe('passwd'); // basename stripped '../../etc/'
+    expect(existsSync(decoy)).toBe(false);   // the contained file was the delete target
   });
 
   it('deletes the uploaded file; subsequent list is empty', async () => {

